scipy/optimize/_optimize.py

```diff
@@ -2116,7 +2116,7 @@ def terminate(warnflag, msg):
         update_l1norm = np.linalg.norm(update, ord=1)
 
     else:
-        if np.isnan(old_fval) or np.isnan(update).any():
+        if np.isnan(old_fval) or np.isnan(update_l1norm):
```

Comment:
... not least because you are also removing the call to `.any()` here.

Reply:
Happy to add the `.any()` back in, but the norm is always going to be a scalar so it's unnecessary. The norm will always be `nan` if any of the elements of the vector (`update`) are `nan`.

I'm not sure about the shape side of this. It looks like previously `update_l1norm` was initialised to `2 * xtol`, and `xtol` is `len(x0) * avextol`. With `len(x0)` and `avextol` being scalars I think this has always just been a float value?